.github/workflows/build.yaml

```diff
@@ -4,41 +4,144 @@ on:
+            compiler: xcode
+            version: "11.3"
     steps:
-      - name: Get source
-        uses: actions/checkout@v2
+      - name: Install dependencies (Linux)
+        if: runner.os == 'Linux'
+        run: |
+          if [ "${{ matrix.config.os }}" = "ubuntu-20.04" ]; then
+            sudo apt-get install -y ninja-build elfutils libzstd-dev
+          else
+            sudo apt-get install -y ninja-build elfutils libzstd1-dev
+          fi
+
+          if [ "${{ matrix.config.compiler }}" = "gcc" ]; then
+            echo "CC=gcc-${{ matrix.config.version }}" >> $GITHUB_ENV
+            echo "CXX=g++-${{ matrix.config.version }}" >> $GITHUB_ENV
+
+            # Disable warnings on ancient compiler.
+            if [ "${{ matrix.config.version }}" = "4.8" ]; then
+              echo "CMAKE_PARAMS=-DWARNINGS_AS_ERRORS=OFF" >> $GITHUB_ENV
+            fi
+
+            sudo apt install -y g++-${{ matrix.config.version }} g++-${{ matrix.config.version }}-multilib
+          else
+            if [ "${{ matrix.config.version }}" = "3.4" ]; then
+              echo "CC=clang" >> $GITHUB_ENV
+              echo "CXX=clang++" >> $GITHUB_ENV
+              echo "CXXFLAGS=-stdlib=libc++" >> $GITHUB_ENV
 
-      - name: Install dependencies (Ubuntu 16 & 18)
-        if: startsWith(matrix.os, 'ubuntu') && matrix.os != 'ubuntu-20.04'
-        run: sudo apt-get install elfutils libzstd1-dev
+              # clang-3.4 is available from trusty
+              sudo add-apt-repository "deb http://dk.archive.ubuntu.com/ubuntu/ trusty main"
+              sudo add-apt-repository "deb http://dk.archive.ubuntu.com/ubuntu/ trusty universe"
+  
+              # Disable warnings on ancient compiler.
+              # doctest does not compile with clang-3.4.
+              echo "CMAKE_PARAMS=-DWARNINGS_AS_ERRORS=OFF -DENABLE_TESTING=OFF" >> $GITHUB_ENV
+            else
+              echo "CC=clang-${{ matrix.config.version }}" >> $GITHUB_ENV
+              echo "CXX=clang++-${{ matrix.config.version }}" >> $GITHUB_ENV
 
-      - name: Install dependencies (Ubuntu 20)
-        if: matrix.os == 'ubuntu-20.04'
-        run: sudo apt-get install elfutils libzstd-dev
+              if [ "${{ matrix.config.version }}" = "3.5" ]; then
+                # Disable warnings on ancient compiler.
+                echo "CMAKE_PARAMS=-DWARNINGS_AS_ERRORS=OFF" >> $GITHUB_ENV
+              fi
+            fi
+
+            sudo apt update
+            sudo apt install -y clang-${{ matrix.config.version }} g++-multilib
+          fi
+
+      - name: Install dependencies (Mac)
+        if: runner.os == 'macOS'
+        run: |
+          brew install ninja
+          if [ "${{ matrix.config.compiler }}" = "gcc" ]; then
+            brew install gcc@${{ matrix.config.version }}
+            echo "CC=gcc-${{ matrix.config.version }}" >> $GITHUB_ENV
+            echo "CXX=g++-${{ matrix.config.version }}" >> $GITHUB_ENV
+          else
+            ls -ls /Applications/
```

Comment:
Debug statement? Maybe remove?

Reply:
Found more xcode versions in that ls statement add added them to the job matrix 😄